lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,778 @@
+                changed = True
+
+    else:
+        try:
+            elb = connection.create_load_balancer(**params)['LoadBalancers'][0]
+            changed = True
+            new_load_balancer = True
+        except ClientError as e:
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+        if module.params.get("wait"):
+            status_achieved, new_elb = wait_for_status(connection, module, elb['LoadBalancerArn'], 'active')
+
+    # Now set ELB attributes. Use try statement here so we can remove the ELB if this stage fails
+    update_attributes = []
+
+    # Get current attributes
+    current_elb_attributes = get_elb_attributes(connection, module, elb['LoadBalancerArn'])
+
+    if access_logs_enabled and current_elb_attributes['access_logs_s3_enabled'] != "true":
+        update_attributes.append({'Key': 'access_logs.s3.enabled', 'Value': "true"})
+    if not access_logs_enabled and current_elb_attributes['access_logs_s3_enabled'] != "false":
+        update_attributes.append({'Key': 'access_logs.s3.enabled', 'Value': 'false'})
+    if access_logs_s3_bucket is not None and access_logs_s3_bucket != current_elb_attributes['access_logs_s3_bucket']:
+        update_attributes.append({'Key': 'access_logs.s3.bucket', 'Value': access_logs_s3_bucket})
+    if access_logs_s3_prefix is not None and access_logs_s3_prefix != current_elb_attributes['access_logs_s3_prefix']:
+        update_attributes.append({'Key': 'access_logs.s3.prefix', 'Value': access_logs_s3_prefix})
+    if deletion_protection and current_elb_attributes['deletion_protection_enabled'] != "true":
+        update_attributes.append({'Key': 'deletion_protection.enabled', 'Value': "true"})
+    if not deletion_protection and current_elb_attributes['deletion_protection_enabled'] != "false":
+        update_attributes.append({'Key': 'deletion_protection.enabled', 'Value': "false"})
+    if idle_timeout is not None and str(idle_timeout) != current_elb_attributes['idle_timeout_timeout_seconds']:
+        update_attributes.append({'Key': 'idle_timeout.timeout_seconds', 'Value': str(idle_timeout)})
+
+    if update_attributes:
+        try:
+            connection.modify_load_balancer_attributes(LoadBalancerArn=elb['LoadBalancerArn'], Attributes=update_attributes)
+            changed = True
+        except ClientError as e:
+            # Something went wrong setting attributes. If this ELB was created during this task, delete it to leave a consistent state
+            if new_load_balancer:
+                connection.delete_load_balancer(LoadBalancerArn=elb['LoadBalancerArn'])
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Now, if required, set ELB listeners. Use try statement here so we can remove the ELB if this stage fails
+    try:
+        listener_changed = create_or_update_elb_listeners(connection, module, elb)
```

Comment:
Not a blocker, but do we want a purge_listeners option? To do any modification to the ELB all the listeners must be specified each time.